app/views/videos/advanced_search.html.haml

```diff
@@ -0,0 +1,51 @@
+%section.advanced_search.container
+  = form_tag advanced_search_videos_path, class: 'form-horizontal', method: :get do
+    %header
+      %h1 Advanced Search
+    .form-group
+      .col-sm-6
+        = search_field_tag :query, params[:query], placeholder: "Search ideos by title, description", class: "form-control"
+    .form-group
+      .col-sm-6
+        .checkbox
+          = label_tag :reviews do
+            = check_box_tag :reviews, true, params[:reviews].present?
+            Include Reviews
+    .form-group
+      .row.average_rating_label Average Rating
+      = label_tag :rating_from, "From", class: "col-sm-1 control-label"
+      .col-sm-2
+        = select_tag :rating_from, options_for_select((10..50).map { |num| num/10.0 }, params[:rating_from]), prompt: "-", class: "form-control"
+      = label_tag :rating_to, "To", class: "col-sm-1 control-label"
+      .col-sm-2
+        = select_tag :rating_to, options_for_select((10..50).map { |num| num/10.0 }, params[:rating_to]), prompt: "-", class: "form-control"
+    .form-group
+      .col-sm-6
+        = button_tag(type: "submit", class: "btn btn-primary") do
+          %span.glyphicon.glyphicon-search
+          Search
+  %hr
+  - if params[:query].present?
+    %h2.results_count Results: #{pluralize(@videos.count, "video")} found
+  - @videos.each do |video|
+    %article.video.row
+      .video_data
+        %figure.col-sm-2
+          %img(src="#{video.small_cover.url}" alt="video cover")
+        .video_body.col-sm-10
+          %header
+            %h3
+              = link_to highlighted_text(video, :title), video_path(video.id)
+          %p.video_description
+            = highlighted_text(video, :description)
+          %hr
+          .details
+            %span.average_rating 
+              Average rating:
+              = video.average_rating
+            %span.reviews_count 
+              = pluralize(video.reviews.count, "review")
+          - if video.reviews.any?
+            %footer.video_review
+              = video.try(:highlight).try("reviews.content") ? video.highlight["reviews.content"][0].html_safe : video.reviews.last.content
```

Comment:
You have `highlight_text` (view) method. Current implementation can not help with the code on line 50, but your view code looks a bit inconsistent now. A solution to this would be to either:

1. build more robuts `highlight_text` method
2. Use a decorator object and have it handle the highlighting logic. You would only call here `video.reviews_content`